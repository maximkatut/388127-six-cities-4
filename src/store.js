import {applyMiddleware, createStore} from 'redux';
import {composeWithDevTools} from "redux-devtools-extension";
import thunk from 'redux-thunk';

import {ActionCreator as DataActionCreator} from './actions/data-actions.js';
import {ActionCreator as UserActionCreator} from './actions/user-actions.js';
import {createAPI} from './api';
import {AppRoute} from './const';
import history from './history';
import rootReducer from './reducers';
import {Operation as DataOperation} from './reducers/data/data-reducer';
import {AuthorizationStatus, Operation as UserOperation} from './reducers/user/user-reducer';

const onUnauthorized = () => {
  store.dispatch(UserActionCreator.requireAuthorization(AuthorizationStatus.NO_AUTH));
  history.push(`${AppRoute.LOGIN}`);
};

const onBadRequest = (err) => {
  store.dispatch(DataActionCreator.catchError(true, err));
};

const api = createAPI(onUnauthorized, onBadRequest);

export const store = createStore(
    rootReducer,
    composeWithDevTools(
        applyMiddleware(thunk.withExtraArgument(api))
    )
);

store.dispatch(DataOperation.loadOffers());
store.dispatch(UserOperation.checkAuth());
