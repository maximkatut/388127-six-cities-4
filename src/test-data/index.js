export const offers = [
  {
    appliences: [`Wi-Fi`, `Microwave`],
    bedrooms: 3,
    coordinates: [52.3809553943508, 4.939309666406198],
    description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`],
    guests: 2,
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Max`,
      super: true
    },
    id: `1`,
    imagesGallery: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`
    ],
    mainImage: `img/apartment-01.jpg`,
    offerType: `House`,
    premium: true,
    price: 200,
    rate: 5,
    title: `Nice, cozy, warm big bed house`
  },
  {
    appliences: [`Wi-Fi`, `Microwave`],
    bedrooms: 4,
    coordinates: [52.3809553943508, 4.939309666406198],
    description: [`An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`],
    guests: 3,
    host: {
      avatar: `img/avatar-max.jpg`,
      name: `Anton`,
      super: false
    },
    id: `2`,
    imagesGallery: [
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`
    ],
    mainImage: `img/apartment-02.jpg`,
    offerType: `Apartment`,
    premium: true,
    price: 300,
    rate: 4,
    title: `Nice, cozy, warm big bed apartment`
  }
];

