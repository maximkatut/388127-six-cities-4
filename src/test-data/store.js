import {cities, offers} from '.';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

export const store = mockStore({
  activeCityName: cities[0].name,
  offers,
  cities
});
