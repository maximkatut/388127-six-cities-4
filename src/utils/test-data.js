export const offerCards = [
  {
    id: `1`,
    imageSrc: `img/apartment-02.jpg`,
    offerType: `Room`,
    premium: false,
    price: 200,
    rate: 5,
    title: `Nice, cozy, warm big bed apartment`
  },
  {
    id: `2`,
    imageSrc: `img/apartment-03.jpg`,
    offerType: `House`,
    premium: true,
    price: 300,
    rate: 4,
    title: `Nice, cozy, warm big bed apartment`
  }
];
