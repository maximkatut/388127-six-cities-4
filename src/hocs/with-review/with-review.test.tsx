import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import Enzyme, {mount} from 'enzyme';
import Adapter from "enzyme-adapter-react-16";

import {store} from '../../test-data/store';
import withReview from './with-review';

Enzyme.configure({
  adapter: new Adapter()
});

const MockComponent = (props) => {
  const {onFormSubmit, onRadioChange, onInputChange, review} = props;

  return (
    <form onSubmit={onFormSubmit}>
      <input onChange={onRadioChange}></input>
      <textarea onChange={onInputChange} value={review}/>
    </form>
  );
};

MockComponent.propTypes = {
  onFormSubmit: PropTypes.func.isRequired,
  onRadioChange: PropTypes.func.isRequired,
  onInputChange: PropTypes.func.isRequired,
  review: PropTypes.string.isRequired
};

const MockComponentWrapped = withReview(MockComponent);

describe(`withReview`, () => {
  it(`withReview is rendered correctly`, () => {
    const postNewReview = jest.fn();
    const component = renderer.create((
      <Provider store={store}>
        <MockComponentWrapped
          offerId={1}
          postNewReview={postNewReview}
          isBusy={false}
        />
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    });

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`withReview should dispatch an operation after form submit`, () => {
    const postNewReview = jest.fn();

    const mockEvent = {
      preventDefault() {
        return undefined;
      }
    };

    const wrapper = mount(
        <Provider store={store}>
          <MockComponentWrapped offerId={1} postNewReview={postNewReview} isBusy={false}/>
        </Provider>
    );

    wrapper.find(`form`).at(0).simulate(`submit`, mockEvent);

    expect(postNewReview).toHaveBeenCalledTimes(1);
  });
});
