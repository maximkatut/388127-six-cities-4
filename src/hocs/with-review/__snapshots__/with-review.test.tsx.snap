// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withReview withReview is rendered correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
  />
  <textarea
    onChange={[Function]}
    value=""
  />
</form>
`;
